internal/cmd/postgresflex/instance/clone/clone_test.go

```diff
@@ -0,0 +1,325 @@
+			flagValues: fixtureRequiredFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, projectIdFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 1",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureRequiredFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 2",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureRequiredFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+		{
+			description: "instance id invalid 1",
+			argValues:   []string{""},
+			flagValues:  fixtureRequiredFlagValues(),
+			isValid:     false,
+		},
+		{
+			description: "instance id invalid 2",
+			argValues:   []string{"invalid-uuid"},
+			flagValues:  fixtureRequiredFlagValues(),
+			isValid:     false,
+		},
+		{
+			description: "recovery timestamp is missing",
+			flagValues: fixtureRequiredFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, recoveryTimestampFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "recovery timestamp is empty",
+			flagValues: fixtureRequiredFlagValues(func(flagValues map[string]string) {
+				flagValues[recoveryTimestampFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "recovery timestamp is invalid",
+			flagValues: fixtureRequiredFlagValues(func(flagValues map[string]string) {
+				flagValues[recoveryTimestampFlag] = "test"
+			}),
+			isValid: false,
+		},
+		{
+			description: "recovery timestamp is invalid 2",
+			flagValues: fixtureRequiredFlagValues(func(flagValues map[string]string) {
+				flagValues[recoveryTimestampFlag] = "11:00 12/12/2024"
+			}),
+			isValid: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			cmd := NewCmd()
+			err := globalflags.Configure(cmd.Flags())
+			if err != nil {
+				t.Fatalf("configure global flags: %v", err)
+			}
+
+			for flag, value := range tt.flagValues {
+				err := cmd.Flags().Set(flag, value)
+				if err != nil {
+					if !tt.isValid {
+						return
+					}
+					t.Fatalf("setting flag --%s=%s: %v", flag, value, err)
+				}
+			}
+
+			err = cmd.ValidateArgs(tt.argValues)
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error validating args: %v", err)
+			}
+
+			err = cmd.ValidateRequiredFlags()
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error validating flags: %v", err)
+			}
+
+			model, err := parseInput(cmd, tt.argValues)
+			if err != nil {
+				if !tt.isValid {
+					return
+				}
+				t.Fatalf("error parsing flags: %v", err)
+			}
+
+			if !tt.isValid {
+				t.Fatalf("did not fail on invalid input")
+			}
+			diff := cmp.Diff(model, tt.expectedModel)
+			if diff != "" {
+				t.Fatalf("Data does not match: %s", diff)
+			}
+		})
+	}
+}
+
+func TestBuildRequest(t *testing.T) {
```

Comment:
This one should have a couple more test cases, testing `nil` storage class and size